packages/cozy-realtime/src/index.js

```diff
@@ -13,28 +13,59 @@ const WEBSOCKET_STATE = {
 const NUM_RETRIES = 3
 const RETRY_BASE_DELAY = 1000
 
+// stored subscriptions arguments to rerun while a retry
+// stored as a Map { [doctype]: socket }
+let subscriptionsState = new Set()
+
+export const getSubscriptionsState = () => subscriptionsState
+
+// listener key computing, according to doctype only or with doc id
+const LISTENER_KEY_SEPARATOR = '/' // safe since we can't have a '/' in a doctype
+const getListenerKey = (doctype, docId) =>
+  docId ? [doctype, docId].join(LISTENER_KEY_SEPARATOR) : doctype
+
+const getTypeAndIdFromListenerKey = listenerKey => {
+  const splitResult = listenerKey.split(LISTENER_KEY_SEPARATOR)
+  return {
+    doctype: splitResult[0],
+    docId: splitResult.length > 1 ? splitResult[1] : null
+  }
+}
+
 // Send a subscribe message for the given doctype trough the given websocket, but
 // only if it is in a ready state. If not, retry a few milliseconds later.
-function subscribeWhenReady(doctype, socket) {
+const MAX_SOCKET_POLLS = 500 // to avoid infinite poling
+export function subscribeWhenReady(
+  doctype,
+  socket,
+  docId,
```

Comment:
[nitpick] I'd have inversed the order of the two first arguments: `function subscribeWhenReady(socket, doctype, docId, remainedTries)` to keep `doctype` and `docId` together